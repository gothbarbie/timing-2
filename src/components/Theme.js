import React from 'react'
import PropTypes from 'prop-types'
import { ThemeProvider } from 'styled-components'

import theme from './themes/default'

const Theme = ({ children }) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
)

Theme.propTypes = {
  children: PropTypes.node,
}

export default Theme
