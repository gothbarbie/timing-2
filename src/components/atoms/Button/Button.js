import React from 'react'
import styled from 'styled-components'

const Default = styled.button`
  cursor: pointer;
  width: ${({ size }) => size === 's' && 'auto'};
  width: ${({ fullWidth }) => fullWidth && '100%'};

  background-color: transparent;
  color: ${({ theme }) => theme.colors.gray};
  color: ${({ active, theme }) => active && theme.colors.primary};
  outline: none;
  font-size: 1.6rem;
  font-weight: 500;
  border: 1px solid transparent;
  border-color: ${({ active, theme }) => active && theme.colors.gray};
  padding: 1rem 2rem;
  padding: ${({ size }) => size === 's' && '.5rem'};

  &:focus {
    border-color: ${({ theme }) => theme.colors.primary};
  }

  &:hover {
    background-color: ${({ theme }) => theme.colors.lightGray};
    color: ${({ theme }) => theme.colors.primary};
    border-color: ${({ theme }) => theme.colors.gray};
  }

  &:active {
    box-shadow: inset 0 2px 2px rgba(0, 0, 0, 0.25);
  }

  border-radius: 4px;
`

const WithIcon = styled(Default)`
  &:focus {
    border-color: ${({ theme }) => theme.colors.lightGray};
    color: ${({ theme }) => theme.colors.primary};
  }

  &:hover {
    border-color: ${({ theme }) => theme.colors.gray};
    color: ${({ theme }) => theme.colors.primary};
    fill: ${({ theme }) => theme.colors.primary};
  }
`

const Primary = styled(Default)`
  background-color: ${({ theme }) => theme.colors.primary};
  color: ${({ theme }) => theme.colors.white};

  box-shadow: ${({ active, theme }) =>
    active ? 'inset 0 2px 2px rgba(0, 0, 0, 0.25)' : 'none'};

  &:focus {
    border-color: ${({ theme }) => theme.colors.white};
  }

  &:hover {
    background-color: ${({ theme }) => theme.colors.primary};
    border-color: ${({ theme }) => theme.colors.white};
    color: ${({ theme }) => theme.colors.white};
  }
`

export const Button = ({
  active = false,
  children,
  fullWidth,
  hasIcon,
  primary,
  size,
  title,
  onClick,
}) => {
  if (primary)
    return (
      <Primary
        active={active}
        fullWidth={fullWidth}
        onClick={onClick}
        size={size}
        title={title}
      >
        {children}
      </Primary>
    )
  if (hasIcon) {
    return (
      <WithIcon
        active={active}
        fullWidth={fullWidth}
        hasIcon={hasIcon}
        onClick={onClick}
        title={title}
      >
        {children}
      </WithIcon>
    )
  }

  return (
    <Default
      active={active}
      fullWidth={fullWidth}
      onClick={onClick}
      size={size}
      title={title}
    >
      {children}
    </Default>
  )
}

/** @component */
export default Button
