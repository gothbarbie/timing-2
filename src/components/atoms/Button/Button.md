# Button

```js
  <Button>Test Button</Button>
```
