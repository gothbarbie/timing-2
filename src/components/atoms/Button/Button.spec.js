import React from 'react'
import { shallow } from 'enzyme'

import { theme } from '../../Theme'
import Button from './Button'

describe('generic/atoms/Button', () => {
  it('renders Default style', () => {
    const component = shallow(<Button theme={theme}>Text</Button>)

    expect(component).toMatchSnapshot()
  })

  it('renders Primary style', () => {
    const component = shallow(
      <Button primary theme={theme}>
        Text
      </Button>
    )

    expect(component).toMatchSnapshot()
  })

  it('renders Icon style', () => {
    const component = shallow(
      <Button icon theme={theme}>
        <i />
      </Button>
    )

    expect(component).toMatchSnapshot()
  })
})
