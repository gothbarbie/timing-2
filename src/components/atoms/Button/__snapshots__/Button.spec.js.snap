// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generic/atoms/Button renders Default style 1`] = `
<styled.button
  active={false}
>
  Text
</styled.button>
`;

exports[`generic/atoms/Button renders Icon style 1`] = `
<styled.button
  active={false}
>
  <i />
</styled.button>
`;

exports[`generic/atoms/Button renders Primary style 1`] = `
<Styled(styled.button)
  active={false}
>
  Text
</Styled(styled.button)>
`;
